lib/publiccloud/azure.pm

```diff
@@ -295,8 +306,8 @@ sub cleanup {
     script_run("az vm boot-diagnostics get-boot-log --ids $id | jq -r '.' > bootlog.txt");
     upload_logs("bootlog.txt", failok => 1);
 
-    $self->SUPER::cleanup();
-    $self->provider_client->cleanup();
+    $self->SUPER::destroy();
```

Comment:
According to the PR description, the `destroy` got introduced to decouple crucial cleanup steps (i.e. `terraform destroy`) from optional cleanup steps like collection and upload of logs.

But this part here couples them again, which omits the purpose of the PR. If `destroy` is also collecting all logs then I don't get why we should introduce a second cleanup routine.